core/capabilities/syncer.go

```diff
@@ -56,26 +71,486 @@ var defaultStreamConfig = p2ptypes.StreamConfig{
 const maxRetryCount = 60
 
 // RegistrySyncer updates local Registry to match its onchain counterpart
-func NewRegistrySyncer(peerWrapper p2ptypes.PeerWrapper, registry core.CapabilitiesRegistry, dispatcher remotetypes.Dispatcher, lggr logger.Logger,
-	networkSetup HardcodedDonNetworkSetup) *registrySyncer {
+func NewRegistrySyncer(
+	peerWrapper p2ptypes.PeerWrapper,
+	registry core.CapabilitiesRegistry,
+	dispatcher remotetypes.Dispatcher,
+	lggr logger.Logger,
+	networkSetup HardcodedDonNetworkSetup,
+	relayer contractReaderFactory,
+	registryAddress string,
+) (*registrySyncer, error) {
+	stopCh := make(services.StopChan)
+	ctx, _ := stopCh.NewCtx()
+	reader, err := newRemoteRegistryReader(ctx, relayer, registryAddress)
+	if err != nil {
+		return nil, err
+	}
+
+	return newRegistrySyncer(
+		stopCh,
+		peerWrapper,
+		registry,
+		dispatcher,
+		lggr,
+		networkSetup,
+		reader,
+	), nil
+}
+
+func newRegistrySyncer(
+	stopCh services.StopChan,
+	peerWrapper p2ptypes.PeerWrapper,
+	registry core.CapabilitiesRegistry,
+	dispatcher remotetypes.Dispatcher,
+	lggr logger.Logger,
+	networkSetup HardcodedDonNetworkSetup,
+	reader reader,
+) *registrySyncer {
 	return &registrySyncer{
+		stopCh:       stopCh,
 		peerWrapper:  peerWrapper,
 		registry:     registry,
 		dispatcher:   dispatcher,
 		networkSetup: networkSetup,
 		lggr:         lggr,
+		reader:       reader,
 	}
 }
 
 func (s *registrySyncer) Start(ctx context.Context) error {
-	s.wg.Add(1)
-	go s.launch(context.Background())
```

Comment:
Let's comment this out for now to still leave an option to run a hardcoded version for local tests. Later we could make it cleaner or remove completely if we have good tooling.

Reply:
Should this be a config option instead? I.e., don't start Syncer if RemoteAddress is not specified. I suspect we'll want to support setups that don't go through the remote registry.